### ISSUE TITLE
latest version of mainline-content-discovery has iroh version 0.13 as a dependency

### ISSUE THREAD

**Issue by philrhc** [n0-computer/iroh-experiments#19]
seems that latest version of mainline-content-discovery is built with iroh version 0.13 which causes some issues when trying to update swarmie to the latest version

https://docs.rs/crate/iroh-mainline-content-discovery/latest/source/Cargo.lock#1485

**Comment by matheus23** [comment]
We should probably try to not upload that lockfile to crates.io. Nevertheless, it shouldn't have an effect if you install the iroh-mainline-content-discovery crate.

The actual iroh versions it's using are specified in its workspace `Cargo.toml`, and they're version 0.28:
https://github.com/n0-computer/iroh-experiments/blob/4e052c6b34720e26683083270706926a84e49411/content-discovery/Cargo.toml#L29-L31

Can you tell us more about the exact error you're getting?

**Comment by philrhc** [comment]
i'm working to bring swarmie back to life - https://github.com/philrhc/swarmie/tree/from-sendme

```
error[E0308]: mismatched types
   --> src/main.rs:588:9
    |
588 |         content, 
    |         ^^^^^^^ expected `iroh_base::hash::HashAndFormat`, found `HashAndFormat`
    |
    = note: `HashAndFormat` and `iroh_base::hash::HashAndFormat` have similar names, but are actually distinct types
note: `HashAndFormat` is defined in crate `iroh_base`
   --> /home/pcummins/.cargo/registry/src/index.crates.io-6f17d22bba15001f/iroh-base-0.28.0/src/hash.rs:225:1
    |
225 | pub struct HashAndFormat {
    | ^^^^^^^^^^^^^^^^^^^^^^^^
note: `iroh_base::hash::HashAndFormat` is defined in crate `iroh_base`
   --> /home/pcummins/.cargo/registry/src/index.crates.io-6f17d22bba15001f/iroh-base-0.13.0/src/hash.rs:209:1
    |
209 | pub struct HashAndFormat {
    | ^^^^^^^^^^^^^^^^^^^^^^^^
    = note: perhaps two different versions of crate `iroh_base` are being used?
```

**Comment by philrhc** [comment]
i don't think that it's been released since the change was made to update the version to 0.28.0, here is the cargo.toml of the crate:

https://docs.rs/crate/iroh-mainline-content-discovery/latest/source/Cargo.toml

**Comment by matheus23** [comment]
Right. Yeah it seems like we haven't made releases for mainline-content-discovery recently.
Thanks for letting us know!
We'll let you know when we do the next release. In the mean time you might be able to get stuff working with git dependencies or `[patch.crates-io]` annotations.
